### PR TITLE
chore: new test harness options

### PR DESCRIPTION
**Pull request by mxschmitt**
This makes the following work:

```xml
<?xml version="1.0" encoding="utf-8"?>
<RunSettings>
  <Playwright>
    <Browser>chromium</Browser>
    <ExpectTimeout>10000</ExpectTimeout>
    <LaunchOptions>
      <Headless>false</Headless>
      <Channel>msedge</Channel>
    </LaunchOptions>
     <!-- The following is not yet implemented -->
    <ActionTimeout>10000</ActionTimeout>
    <Retries>3</Retries>
    <Trace>OnFirstRetry</Trace>
  </Playwright>
</RunSettings>
```

instead of `TestRunParameters` which gives us more flexibility for the future and is easier to use when overriding it via CLI args. We are aware that we might break users, but we didn't announce it yet so it should not harm too many.

### CODE REVIEW

File: src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+                return BrowserType.Chromium;
+            }
+            private set { }
+        }
+
+        public static float? ExpectTimeout
+        {
+            get
+            {
+                if (_settings == null)
```
Comment (mxschmitt) [inline src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs:63]: in ALL the getters `_settings` can be null if there is no `RUNSETTINGS` file given. We fallback then to defaults.

File: src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+                browserName != BrowserType.Webkit)
+            {
+                throw new ArgumentException($"Invalid browser name: {browserName}");
+            }
+        }
+
+        public void Load(XmlReader reader)
+        {
+            _settings = new PlaywrightSettingsXml(reader);
+        }
```
Comment (kblok) [inline src/Playwright.TestAdapter/PlaywrightSettingsProvider.cs:107]:
```suggestion
            => _settings = new PlaywrightSettingsXml(reader);
```

File: src/Playwright.TestAdapter/PlaywrightSettingsXml.cs [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+                        break;
+                    default:
+                        Console.Error.WriteLine($"Playwright RunSettings Parsing Error: Playwright>{reader.Name} is not implemented");
+                        break;
+                }
+            }
+        }
+
+        private static T ParseXmlIntoClass<T>(XmlReader reader)
+        where T : class, new()
```
Comment (kblok) [inline src/Playwright.TestAdapter/PlaywrightSettingsXml.cs:72]:
```suggestion
        private static T ParseXmlIntoClass<T>(XmlReader reader) where T : class, new()
```

File: src/Playwright.TestAdapter/PlaywrightSettingsXml.cs [hunk 4]
```diff
@@ -0,0 +1,147 @@
...
+        {
+            var property = options.GetType().GetProperty(key);
+            if (property == null)
+            {
+                return;
+            }
+            var type = Nullable.GetUnderlyingType(property.PropertyType) ?? property.PropertyType;
+            switch (type)
+            {
+                case Type t when t == typeof(string):
```
Comment (kblok) [inline src/Playwright.TestAdapter/PlaywrightSettingsXml.cs:97]:
```
case string stringValue:
case bool boolValue:
```
Won't work?

Comment (mxschmitt) [reply src/Playwright.TestAdapter/PlaywrightSettingsXml.cs:98]: compile does not allow it.

File: src/Playwright.TestAdapter/Playwright.TestAdapter.csproj [hunk 5]
```diff
@@ -0,0 +1,18 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <TargetFramework>netstandard2.0</TargetFramework>
+    <Nullable>enable</Nullable>
+    <ReleaseVersion>0.0.0</ReleaseVersion>
```
Comment (yury-s) [inline src/Playwright.TestAdapter/Playwright.TestAdapter.csproj:6]: is it expected to be all zeros?

File: src/Playwright.TestAdapter/PlaywrightSettingsXml.cs [hunk 6]
```diff
@@ -0,0 +1,147 @@
...
+                    continue;
+                }
+                switch (reader.Name)
+                {
+                    case "Browser":
+                        reader.Read();
+                        Browser = reader.Value;
+                        break;
+                    case "LaunchOptions":
+                        LaunchOptions = ParseXmlIntoClass<BrowserTypeLaunchOptions>(reader);
```
Comment (yury-s) [inline src/Playwright.TestAdapter/PlaywrightSettingsXml.cs:54]: Shouldn't this go under Browser section?

File: src/Playwright.TestAdapter/PlaywrightSettingsXml.cs [hunk 7]
```diff
@@ -0,0 +1,147 @@
...
+                        break;
+                    case "Headless":
+                        reader.Read();
```
Comment (yury-s) [inline src/Playwright.TestAdapter/PlaywrightSettingsXml.cs:57]: Ditto?

**Review by yury-s — APPROVED** [review]

File: src/Playwright.TestAdapter/Playwright.TestAdapter.csproj (hunk 5 above)
Comment (mxschmitt) [reply src/Playwright.TestAdapter/Playwright.TestAdapter.csproj:15]: yes and fixed.

**Pull request merged**